geopandas/sindex.py

```diff
@@ -2,10 +2,22 @@
 from collections import namedtuple
 
 import pandas as pd
+import numpy as np
 
 from . import _compat as compat
 
 
+VALID_QUERY_PREDICATES = {
+    None,
+    "intersects",
+    "within",
+    "contains",
+    "overlaps",
+    "crosses",
+    "touches",
+}
+
+
```

Comment:
I think this could be a good first step towards defining what we expect a generic `sindex` API to look like. `sindex` classes can adopt these defaults or a custom set of supported predicates, and what I expect will eventually be generic/abstract base class code will catch errors appropriately.

Reply:
Let's keep the discussion about a generic base class in https://github.com/geopandas/geopandas/issues/1344